packages/mui-base/src/Accordion/Header/AccordionHeader.tsx

```diff
@@ -0,0 +1,63 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { BaseUIComponentProps } from '../../utils/types';
+import { useComponentRenderer } from '../../utils/useComponentRenderer';
+import type { AccordionItem } from '../Item/AccordionItem';
+import { useAccordionItemContext } from '../Item/AccordionItemContext';
+import { accordionStyleHookMapping } from '../Item/styleHooks';
+
+/**
+ *
+ * Demos:
+ *
+ * - [Accordion](https://base-ui.netlify.app/components/react-accordion/)
+ *
+ * API:
+ *
+ * - [AccordionHeader API](https://base-ui.netlify.app/components/react-accordion/#api-reference-AccordionHeader)
+ */
+const AccordionHeader = React.forwardRef(function AccordionHeader(
+  props: AccordionHeader.Props,
+  forwardedRef: React.ForwardedRef<HTMLHeadingElement>,
+) {
+  const { render, className, ...otherProps } = props;
+
+  const { ownerState } = useAccordionItemContext();
+
+  const { renderElement } = useComponentRenderer({
+    render: render ?? 'h3',
```

Comment:
> Each accordion header button is wrapped in an element with role [heading](https://w3c.github.io/aria/#heading) that has a value set for [aria-level](https://w3c.github.io/aria/#aria-level) that is appropriate for the information architecture of the page

@michaldudak According to the [aria pattern](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/), `role="heading"` should be required on this subcomponent, which requires the use of `aria-level` as well - I suppose the aria heading level is the arbitrary part, but using `<div role="heading">` without an explicit `aria-level` will default it to level `2` anyway! So I used `h3` because the [example](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/examples/accordion/) does so

However, if we default to a non-heading element like `div` and expose an `aria-level` prop instead, it might make this more visible to users 🤔 

`<Accordion.Heading render={<h2 />} />` vs `<Accordion.Heading aria-level={2} />`

What do you think? @colmtuite @vladmoroz

Reply:
I'd prefer the original `h3` default. It's easier to explain that they are supposed to render a heading element than get into aria-role and aria-level details.